service/client.go

```diff
@@ -234,10 +285,62 @@ func (c *Client) msgReader() {
 			continue
 		}
 
+		if cm.Type == ClientMessageHello {
+			data, ok := cm.Data.(map[string]string)
+			if ok && data["connID"] != "" {
+				c.mut.Lock()
+				c.connID = data["connID"]
+				c.mut.Unlock()
+			}
+		}
+
 		select {
 		case c.receiveCh <- cm:
 		default:
 			c.sendError(fmt.Errorf("failed to send client message: channel is full"))
 		}
 	}
 }
+
+func (c *Client) reconnectHandler() {
+	var attempt int
+	var waitTime time.Duration
+	for {
+		attempt++
+		if waitTime < maxReconnectInterval {
+			waitTime += c.cfg.ReconnectInterval
+		}
+		time.Sleep(waitTime)
```

Comment:
Does this mean we'll wait a minimum 2 seconds after being disconnected?  I wonder if that's excessive, at least for the first attempt? Should we try to reconnect almost immediately?

Reply:
Attempting a reconnect in the same millisecond is unlikely to work given either the network is down or the service restarted/crashed so I think some wait time makes sense. I made the interval configurable so we can definitely lower it on the other side if we want to.